### ISSUE TITLE
Cant send keys or receive keys, nor can I end the session.

### ISSUE THREAD

**Issue by dubyadud** [SilenceIM/Silence#571]
### How to reproduce
Not known, only way I can potentially recreate this is if I give the certain number out, which if requested I can give.

**Actual result:** explained above
**Expected result:** successful key transfer

### Screenshots
Will edit in screenshots, somewhat worthlesd though in this issue.

### Device info
- Device: Samsung Galaxy S5
- Android version: 7.1
- Silence version and build ID: 47bea30

### Link to debug log
https://gist.github.com/93ad50d3fc32dd1660cf086893b3186d

**Comment by dubyadud** [comment]
I accidentally commented out the description so here it is:

After person A or B tries sending a key to me or vise versa, I can no longer send messages.  All my messages are considered pending and the keys have not been confirmed.  I cant end the session either, so I can not send nessages from these people.

**Comment by joshp23** [comment]
Same issue

Here is debug log 

https://gist.github.com/81114d4ea55e66398640248c7df00154

**Comment by joshp23** [comment]
uninstall and reinstall allowed me to connect to contacts that had initiated private sessions, but still unable to initiate or participate in a private session at all.

**Comment by tuxayo** [comment]
I might have a similar issue with on contact as well.

It seems that the session is successfully initiated for the other person. But not on my side. Thus I only get "Message encrypted fir non-existing session"

**Comment by dubyadud** [comment]
Still not fixed

**Comment by dubyadud** [comment]
unresolved

**Comment by licaon-kter** [comment]
Still? With both 0.15.13 and 0.16.9?

**Comment by dubyadud** [comment]
Yes.  But it's possible I have a device issue also.  The SMSC config is apparently lost when using a custom rom on the S5, and I have a feeling it is contributing to this problem.  I am unable to send long sms, but mms sends fine.  I wonder if the other members in this issue are also using an S5, since it seems (through a bit of googling) that unfortunately the S5 is the nearly the only device that does this.  If there was maybe a way to split the key across multiple messages, it would work great.

**Comment by jdidkdnduisk** [comment]
Same issue here.

This bug might be dependent on ROM.
Can't exchange keys (due to long SMS not sent) on LineageOS 14.1 (Galaxy S7).
But can exchange keys on CyanogenMod 11 (Galaxy S4).
